src/engine/SCons/Tool/jar.py

```diff
@@ -87,10 +89,115 @@ def jarFlags(target, source, env, for_signature):
             break
     return jarflags
 
+def Jar(env, target = None, source = [], *args, **kw):
+    """
+    A pseudo-Builder wrapper around the separate Jar sources{File,Dir}
+    Builders.
+    """
+
+    # jar target should not be a list so assume they passed
+    # no target and want implicit target to be made and the arg
+    # was actaully the list of sources
+    if SCons.Util.is_List(target):
```

Comment:
In this case I'm thinking a warning should be issued rather than quietly discarding the target specified by the user?

Reply:
I agree, warnings for when things are getting weird is good, I'll update with a commit on this request soon.

Also on this note, is there any expected behavior when you pass multiple targets to the Jar builder? I didn't see anything in the man page. In this case, I assumed this behavior from the Java multi-step.py test, where passes only a list of sources to the Jar builder: https://github.com/SConsProject/scons/blob/a583f043aec89e58bf4ab0ab20cc039299eff1df/test/Java/multi-step.py#L377

Reply:
Hopefully expected behavior is covered in the docs and/or the test suite.

Probably worth putting some notice that you'll be dropping the target list in the release notes (src/CHANGES.txt)?

Do any of the tests hit this case?

Reply:
I'm trying to roll out 3.0.1 and would like to get this change in. Any chance you can do the above today?

Reply:
>Hopefully expected behavior is covered in the docs and/or the test suite.

The man page for Jar says it will build "a" jar archive so I assume that implies a single target:
http://scons.org/doc/HTML/scons-man.html
```
Jar() , env.Jar()
Builds a Java archive (.jar) file from the specified list of sources.
```
>Probably worth putting some notice that you'll be dropping the target list in the release notes (src/CHANGES.txt)?

I will update the CHANGES.txt about the target list not being used.

>Do any of the tests hit this case?

No tests test multiple targets passed to Jar, I can write a test for that but I still am not sure what is expected in that case. Currently it assumes if your first argument is a list (base builder class first arg is target), then you meant implicit target and just passed the list of sources, so if someone was trying to pass a list of targets and a list of sources it would ignore the sources and assumes your targets were the sources. Maybe this case should be an Error and stop the build?

>Any chance you can do the above today?

I can update regarding the warning message and the CHANGES.txt in a few hours.